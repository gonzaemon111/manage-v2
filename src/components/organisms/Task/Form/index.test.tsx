import { composeStories } from "@storybook/react";
import { render } from "@testing-library/react";
import * as Stories from "./index.stories";

describe("Organisms/Task/Form コンポーネント", () => {
  describe("Openな時", () => {
    test("適切なclassNameが適用されている", () => {
      const { Default } = composeStories(Stories);
      const { getByText } = render(<Default />);
      expect(getByText("タスク名")).toBeTruthy();
      expect(getByText("終了日時")).toBeTruthy();
      expect(getByText("締切")).toBeTruthy();
      expect(getByText("メモ")).toBeTruthy();
      expect(getByText("送信")).toBeTruthy();
    });
  });

  describe("スナップショットテスト", () => {
    const testCases = Object.values(composeStories(Stories)).map((Story) => [
      Story.storyName,
      Story,
    ]);

    test.each(testCases)("Renders %s story", async (storyName, Story) => {
      if (!Story) {
        throw new Error("Story not Found");
      }

      const tree = render(<Story />);
      expect(tree).toMatchSnapshot();
    });
  });
});
