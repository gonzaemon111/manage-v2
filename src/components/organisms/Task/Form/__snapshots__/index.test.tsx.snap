// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/Task/Form コンポーネント スナップショットテスト Renders Default story 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-8 bg-white dark:bg-slate-700 text-slate-800 dark:text-slate-50 shadow"
      >
        <form
          class="flex flex-col space-y-4 undefined"
        >
          <div
            class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
          >
            <label
              class="text-left md:text-right undefined"
              for="name"
            >
              タスク名
            </label>
            <div
              class="col-span-4"
            >
              <input
                class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
                id="name"
                name="name"
              />
            </div>
          </div>
          <div
            class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
          >
            <label
              class="text-left md:text-right undefined"
              for="finishedAt"
            >
              終了日時
            </label>
            <div
              class="col-span-4"
            >
              <input
                class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
                id="finishedAt"
                name="finishedAt"
                type="datetime-local"
              />
            </div>
          </div>
          <div
            class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
          >
            <label
              class="text-left md:text-right undefined"
              for="deadline"
            >
              締切
            </label>
            <div
              class="col-span-4"
            >
              <input
                class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
                id="deadline"
                name="deadline"
                type="datetime-local"
              />
            </div>
          </div>
          <div
            class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
          >
            <label
              class="text-left md:text-right undefined"
              for="memo"
            >
              メモ
            </label>
            <textarea
              class="col-span-4 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
              id="memo"
              name="memo"
            />
          </div>
          <div
            class="flex justify-center items-center m-8 mb-0"
          >
            <button
              class="text-white bg-slate-300 border-slate-500 opacity-95 font-medium text-sm px-10 py-2.5 text-center shadow-md inline-flex items-center gap-2 mr-2 mb-2"
              disabled=""
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="h-4 w-4"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6 12L3.269 3.126A59.768 59.768 0 0121.485 12 59.77 59.77 0 013.27 20.876L5.999 12zm0 0h7.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
              送信
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-8 bg-white dark:bg-slate-700 text-slate-800 dark:text-slate-50 shadow"
    >
      <form
        class="flex flex-col space-y-4 undefined"
      >
        <div
          class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
        >
          <label
            class="text-left md:text-right undefined"
            for="name"
          >
            タスク名
          </label>
          <div
            class="col-span-4"
          >
            <input
              class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
              id="name"
              name="name"
            />
          </div>
        </div>
        <div
          class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
        >
          <label
            class="text-left md:text-right undefined"
            for="finishedAt"
          >
            終了日時
          </label>
          <div
            class="col-span-4"
          >
            <input
              class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
              id="finishedAt"
              name="finishedAt"
              type="datetime-local"
            />
          </div>
        </div>
        <div
          class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
        >
          <label
            class="text-left md:text-right undefined"
            for="deadline"
          >
            締切
          </label>
          <div
            class="col-span-4"
          >
            <input
              class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
              id="deadline"
              name="deadline"
              type="datetime-local"
            />
          </div>
        </div>
        <div
          class="grid grid-cols-1 md:grid-cols-5 gap-4 items-center justify-center undefined"
        >
          <label
            class="text-left md:text-right undefined"
            for="memo"
          >
            メモ
          </label>
          <textarea
            class="col-span-4 bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-none focus:ring-offset-0 focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 undefined"
            id="memo"
            name="memo"
          />
        </div>
        <div
          class="flex justify-center items-center m-8 mb-0"
        >
          <button
            class="text-white bg-slate-300 border-slate-500 opacity-95 font-medium text-sm px-10 py-2.5 text-center shadow-md inline-flex items-center gap-2 mr-2 mb-2"
            disabled=""
            type="submit"
          >
            <svg
              aria-hidden="true"
              class="h-4 w-4"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 12L3.269 3.126A59.768 59.768 0 0121.485 12 59.77 59.77 0 013.27 20.876L5.999 12zm0 0h7.5"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
            送信
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
